addons/mail/static/src/discuss/core/common/thread_model_patch.js

```diff
@@ -255,19 +268,24 @@ const threadPatch = {
             return undefined;
         }
         const correspondents = this.correspondents;
-        if (correspondents.length === 1) {
+        if (correspondents?.length === 1) {
             // 2 members chat.
             return correspondents[0];
         }
-        if (correspondents.length === 0 && this.channel_member_ids.length === 1) {
+        if (correspondents?.length === 0 && this.channel?.channel_member_ids.length === 1) {
             // Self-chat.
-            return this.channel_member_ids[0];
+            return this.channel?.channel_member_ids[0];
         }
         return undefined;
     },
     /** @returns {import("models").ChannelMember[]} */
     get correspondents() {
-        return this.channel_member_ids.filter(({ persona }) => persona?.notEq(this.store.self));
+        if (!this.channel?.channel_member_ids) {
+            return [];
+        }
```

Comment:
only check on channel should be necessary, many relation should always be set
```suggestion
        if (!this.channel) {
            return [];
        }
```